distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Managed_Resource.enso

```diff
@@ -58,13 +67,19 @@ type Managed_Resource
        ADVANCED
 
        Executes the provided action on the resource managed by the managed
-       resource object.
+       resource object. The action is only invoked if the managed resource
+       has not yet been finalized.
 
        Arguments:
        - action: The action that will be applied to the resource managed by
          resource.
-    with : (Any -> Any) -> Any
-    with self ~action = @Builtin_Method "Managed_Resource.with"
+       - on_missing: Optional action to perform when the value is no longer
+         reachable
+       Returns:
+       Value returned from the `action` or `Nothing` if the managed resource
+       was already finalized
+    with : (Any -> Any) -> Any -> Any
+    with self ~action ~on_missing=Nothing = with_builtin self action on_missing...
```

Comment:
> perhaps even a panic because it is an internal error.

It doesn't have to be an internal error in my opinion. Certainly in the case of _system resources_ one can access `with` which no longer references any value.

Reply:
These kinds of resources are not meant to be used by GUI users, perhaps only library developers and advanced power users.

When using a resource that is known to disappear, we usually should have some logic expecting it. If we don't have it and it disappears, that is IMO a bug.

A dataflow can be easily dropped if returned from a side-effecting computation whose result is discarded. A panic will be more visible and clearly tell us there is a bug. Plus it will have a better stack trace.

Reply:
Since 873bac3 we produce `DataflowError`

Reply:
> dataflow can be easily dropped if returned from a side-effecting computation whose result is discarded

Up to you _lib guys_. So far I see @GregoryTravis's approval (also for `DataflowError`).

Reply:
Looking at current usages of `Uninitialized_State` error - all of them are reported as panics. So if anything, we should stick to being consistent.

Reply:
I agree with a Panic here -- mistaking Nothing for a missing value is a common error, so user code should be strongly encouraged to handle a missing value.

Reply:
So @JaroslavTulach can we change it to a panic? I think we have agreement in the libs team that we'd like to be a Panic as it is less error-prone and consistent with other usages of `Uninitialized_State` (while at it, I'm not sure we should be re-using the `State`-related error, IMO ideally it should be a separate error type, or perhaps just `Illegal_State` error with custom message).

Reply:
I should have read the outcome of this conversation sooner than waking up and eagerly implementing ca7592b ... Don't you like the idea of calling `with` with an error ;-?

Reply:
> Don't you like the idea of calling with with an error ;-?

If not, then this would be the [next commit](https://github.com/JaroslavTulach/enso/commit/e772a532e47268f235c47773157ff468b4329dd1).

Reply:
Integrated with

> [ca7592b](https://github.com/enso-org/enso/commit/ca7592b1ff48364a8ee4aeb60240bc94e0e92b47) ... the idea of calling `with` action callback with an error

- work on _not hiding_ the error continues in #11777